### PR TITLE
feat: Add user statistics api with influxdb.

### LINKED ISSUE #160: Feature: Add record from openim callback.
### Checklist

- [X] I've searched for similar issues and couldn't find anything matching
- [X] I've discussed this feature request in the [OpenIMSDK Slack](https://join.slack.com/t/openimsdk/shared_invite/zt-1tmoj26uf-_FDy3dowVHBiGvLk9e5Xkg) and got positive feedback

### Is this feature request related to a problem?

❎ No

### Problem Description

_No response_

### Solution Description

1. Add user Online and offline callback to get online staffs, used for online staff dispatch.
2. Add message record from callback, used for workload statistics.

### Benefits

Enhance system.

### Potential Drawbacks

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by IRONICBo**
#### 🔍 What type of PR is this?

/kind feature

#### 👀 What this PR does / why we need it:

Add influxdb operation for user statistics.

#### 🅰 Which issue(s) this PR fixes:

Fixes #160

### CODE REVIEW

File: server/internal/api/user_statistics.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	if err != nil {
+		response.FailWithCode(common.UNAUTHORIZED, c)
+
+		return
+	}
+
+	var params requestparams.GetUsersStatisticsParams
+	err = c.ShouldBindJSON(&params)
+	if err != nil {
+		response.FailWithCode(common.INVALID_PARAMS, c)
```
Comment (cubxxw) [inline server/internal/api/user_statistics.go:48]: Use log printing?

File: server/internal/api/user_statistics.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+			return
+		}
+	case string(requestparams.USER_STATISTICS_MESSAGE_COUNT):
+		result, err = svc.ReadUserSendMsgCountPerDay(uuid, params.StartTimestamp, params.EndTimestamp)
+		if err != nil {
+			response.FailWithData(err.Error(), c)
+
+			return
+		}
+	case string(requestparams.USER_STATISTICS_MESSAGE_LENGTH):
```
Comment (cubxxw) [inline server/internal/api/user_statistics.go:71]: If there are other possibilities, add a default option |<
```
func getStatistics(svc *service.UserStatisticService, statisticType string, uuid string, start, end int64) ([]*responseparams.UserStatisticItem, error) {
    switch statisticType {
    case string(requestparams.USER_STATISTICS_ONLINE_TIME):
        return svc.ReadUserOnlineTimePerDay(uuid, start, end)
...
    default:
        return nil, fmt.Errorf("Unsupported statistic type: %s", statisticType)
    }
```

File: server/internal/dal/dao/user_statistic_dao.go [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+const (
+	USER_ACTION_MEASUREMENT  UserMeasurement = "user_action"
+	USER_MESSAGE_MEASUREMENT UserMeasurement = "user_message"
+)
+
+// UserAction user action.
+type UserAction string
+
+// tags.
+const (
```
Comment (cubxxw) [inline server/internal/dal/dao/user_statistic_dao.go:43]: https://juejin.cn/s/golang%20%E5%B8%B8%E9%87%8F%E5%91%BD%E5%90%8D%E8%A7%84%E5%88%99

File: server/internal/dal/dao/user_statistic_dao.go [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+	"github.com/openimsdk/openkf/server/internal/utils"
+	"github.com/openimsdk/openkf/server/pkg/log"
+
+	influxdb2 "github.com/influxdata/influxdb-client-go/v2"
+)
+
+// UserMeasurement user measurement.
+type UserMeasurement string
+
+// measurements.
```
Comment (cubxxw) [inline server/internal/dal/dao/user_statistic_dao.go:33]: what does that mean？

File: server/internal/dal/dao/user_statistic_dao.go [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+	field := "data"
+	// process result to map
+	dataList := make([]*TimeItem, 0)
+
+	query := d.influxClient.BuildQuery(bucket, measurementStr, field, aggPeriod, aggFn, actionStr, uid, startTime, endTime)
+	result, err := d.influxClient.GetQueryAPI().Query(context.Background(), query)
+	if err != nil {
+		log.Errorf("influxClient.GetQueryAPI().Query", "Can not query data: %s", err.Error())
+
+		return dataList, err
```
Comment (cubxxw) [inline server/internal/dal/dao/user_statistic_dao.go:138]: not quite understand `return dataList`

File: server/internal/api/user_statistics.go (hunk 1 above)
Comment (IRONICBo) [reply server/internal/api/user_statistics.go:48]: I will update `FailWithCode ` in other PR with log embeded.

File: server/internal/dal/dao/user_statistic_dao.go (hunk 4 above)
Comment (IRONICBo) [reply server/internal/dal/dao/user_statistic_dao.go:33]: It's base structure in `influxdb`.

**Review by cubxxw — APPROVED** [review]

**Pull request merged**